packages/grid/_modules_/grid/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -332,5 +332,5 @@ export const useGridColumnResize = (apiRef: GridApiRef) => {
   useGridApiEventHandler(apiRef, GRID_COLUMN_RESIZE_STOP, handleResizeStop);
 
   useGridApiOptionHandler(apiRef, GRID_COLUMN_RESIZE, options.onColumnResize);
-  useGridApiOptionHandler(apiRef, GRID_COLUMN_RESIZE_COMMITTED, options.onColumnResizeCommitted);
+  useGridApiOptionHandler(apiRef, GRID_COLUMN_WIDTH_CHANGE, options.onColumnResizeCommitted);
```

Comment:
This looks inconsistent in 3 different ways:

1. `GRID_COLUMN_RESIZE` first, then `GRID_COLUMN_WIDTH_CHANGE` last.
2. `GRID_COLUMN_WIDTH_CHANGE` that fires `props.onColumnResizeCommitted()`
3. There are no events that are fired when the width change because of `flex: 1` and the container width change

Reply:
> I'm confused. Why do we have GRID_COLUMN_HEADER_KEYDOWN but GRID_COLUMN_HEADER_NAVIGATION_KEY_DOWN (KEY_DOWN vs. KEYDOWN)?

Fixed.

> 1. GRID_COLUMN_RESIZE fire first, then GRID_COLUMN_WIDTH_CHANGE last.

I think it would be better to remove `GRID_COLUMN_WIDTH_CHANGE` because it's only used to call `onColumnResizeCommitted`.

> 2. GRID_COLUMN_WIDTH_CHANGE that fires props.onColumnResizeCommitted()

I would open another PR to rename this prop.

> 3. There are no events that are fired when the width change because of flex: 1 and the container width change

The new column width after changing the size of the container can be retrieved by subscribing to `columnsChange`.

Reply:
> I think it would be better to remove GRID_COLUMN_WIDTH_CHANGE because it's only used to call onColumnResizeCommitted

I don't follow. Do you mean removing the `onColumnResizeCommitted` prop?

> I would open another PR to rename this prop.

Won't this only move the problem somewhere else, making the new name of `onColumnResizeCommitted` inconsistent with `onColumnResize`?

> The new column width after changing the size of the container can be retrieved by subscribing to columnsChange.

I assume `columnsChange` also fire when we are finished to resize the columns. Which suggests that we don't need a `GRID_COLUMN_WIDTH_CHANGE` event (I mean, the part on the width changed).

Reply:
`GRID_COLUMN_WIDTH_CHANGE` could be removed as well as the `onColumnResizeCommitted` prop. We fix `GRID_COLUMN_RESIZE` to be called in the same conditions of the removed event . We do these things in another PR.

> I assume columnsChange also fire when we are finished to resize the columns. Which suggests that we don't need a GRID_COLUMN_WIDTH_CHANGE event (I mean, the part on the "width changed").

Yes, the most reliable is `columnsChange`. We also don't need `GRID_COLUMN_RESIZE_START` or `GRID_COLUMN_RESIZE_STOP`. These events are only used inside this hook, where their handler could be called directly.